### PR TITLE
Ifpack2/MueLu: Improving Tests

### PR DESCRIPTION
**Pull request by csiefer2**
Tests: RHEL7; new tests added

Issue: #9200 

Stakeholder feedback: n/a

### CODE REVIEW

File: packages/muelu/test/unit_tests/Smoothers/Ifpack2Smoother.cpp [hunk 1]
```diff
@@ -592,6 +592,61 @@ namespace MueLuTests {
...
+      ifpack2Params.set("partitioner: type","linear");
+      ifpack2Params.set("partitioner: PDE equations",3);
+      
+      Ifpack2Smoother smoother("BLOCKRELAXATION",ifpack2Params);
+      
+      Level level; TestHelpers::TestFactory<SC,LO,GO,NO>::createSingleLevelHierarchy(level);
+      level.Set("A", A);
+      smoother.Setup(level);
+
+      TEST_EQUALITY(1,1);
```
Comment (cgcgcg) [inline packages/muelu/test/unit_tests/Smoothers/Ifpack2Smoother.cpp:618]: Why?

Comment (csiefer2) [reply packages/muelu/test/unit_tests/Smoothers/Ifpack2Smoother.cpp:618]: "Does not segfault in setup"

File: packages/muelu/test/unit_tests/Smoothers/Ifpack2Smoother.cpp [hunk 2]
```diff
@@ -592,6 +592,61 @@ namespace MueLuTests {
...
```
Comment (GrahamBenHarper) [inline packages/muelu/test/unit_tests/Smoothers/Ifpack2Smoother.cpp:647]: Same here

**Review by cgcgcg — APPROVED** [review]

**Comment by GrahamBenHarper** [comment]
I'll add a comment based on our conversation in case we need it for later reference: The `TEST_EQUALITY(1,1);` statements are copied from tests directly above. These tests are testing to make sure the process doesn't crash, and supposedly the original reason for the `TEST_EQUALITY(1,1);` statements is because there has to be some sort of `TEST` in order for things to work properly.

**Comment by csiefer2** [comment]
@william76 's terrible diagram!

**Comment by jhux2** [comment]
@trilinos/framework This PR merged despite the last round of testing having failed.  This shouldn't be possible, right?

**Comment by prwolfe** [comment]
Correct - @allevin, could this be related to the branch we are using?

**Comment by jwillenbring** [comment]
> Correct - @allevin, could this be related to the branch we are using?

@prwolfe Has the dev->master testing been moved to the new branch also?

**Comment by prwolfe** [comment]
> > Correct - @allevin, could this be related to the branch we are using?
> 
> @prwolfe Has the dev->master testing been moved to the new branch also?

No, that is still running master.

**Comment by allevin** [comment]
@prwolfe @jhux2 @jwillenbring This took a while to figure out.   

The last commit of this PR (d3730e3) is the part of the commit chain for PR #9371, when 9371 was merged, Github also indicated that the last sha of 9210 was also merged (since it is part of 9371) at the exact same timestamp.  

What added to the confusion is that the user trilinos-autotester is the avatar that the autotester application uses to perform the merge function.  So it makes it seem like the autotester application performed the merge of 9210 where actually it was PR 9371, which actually merged by the autotester appliction, that merged the last commit of 9210.

Hat tip to @william76 for pointing out what happened here...

**Pull request merged**